WordPress-iOS Edition [Do Not Merge]

kean opened the pull request:
### Description

This branch contains an archived version of WordPressKit designed to be used exclusively by [WordPress-iOS](https://github.com/wordpress-mobile/WordPress-iOS). Consider it a release branch.

This version of WordPressKit is distributed using SwiftPM as a binary target. It has no dependencies to `WordPressShared` and `UIDeviceIdentifier`, and it imports `wpxmlrpc` and `NSObject_SafeExtensions` as `@_implementationOnly`, so there are not exposed in the public interface.

### Updating WordPressKit

On an off chance that the WordPressKit version used by WordPress-iOS needs a patch, please submit it by targeting this branch.

#### Deployment Instructions

The deployment currently requires a few manual steps, but we don't expect it to be needed often.

- Run `.buildkite/create-xcframeworks.sh` to create binaries
- Upload `.build/xcframeworks/WordPressKit.zip` to this repo
- Update binary target in `Package.swift` with a new URL and checksum*
- Push the new changes to your branch
- (After testing is done) Merge changes in `wpios-edition`

> checksum is echoed at the end of the create-xcframeworks.sh script

kean committed:
Remove WordPressShared dependency

kean committed:
Inline WordPressShared

kean committed:
Update unit tests

kean committed:
Remove UIDeviceIdentifier dependency

kean committed:
Link NSObject-SafeExpectations and wpxmlrpc statically

kean committed:
Add a script to build xcframework

kean committed:
(Test) import dependencies using @_implementationOnly

kean committed:
Prefix extensions

kean committed:
Fix missing access control modifier

kean committed:
Remove Alamofire dependency from tests

kean committed:
Deintegrate CocoaPods

kean committed:
Remove unused files

kean committed:
Add SwiftLint back

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Package.swift</code> was...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean committed:
Add SwiftLint phase and fix violations

kean committed:
Remove validate-podspec phase

kean committed:
Fix Podfile

dangermattic commented on Podfile:
```diff
@@ -8,38 +8,14 @@ use_frameworks!
 APP_IOS_DEPLOYMENT_TARGET = Gem::Version.new('13.0')
 
 platform :ios, APP_IOS_DEPLOYMENT_TARGET
+workspace "./WordPressKit.xcworkspace"
```
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPressKit-iOS/blob/24168ae77b4bb0e20553c5ed86121f9dd76cdf8a/Podfile#L11"></span>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

kean committed:
Add changes made in wpios

kean commented:
Versions:

- **v0.0.1**: https://github.com/user-attachments/files/16200443/WordPressKit.zip (09fde69ac4ca044a02ed52daa478c025c66fcfbf6cb5b64af2ab7959fc403508)

kean committed:
Add Package.swift

kean committed:
Update some of the public APIs

kean committed:
Add is_wpcom_staging_site

kean committed:
Add missing @frozen

kean committed:
Update WPKit (#817)

kean committed:
Update Package.swift

spencertransier committed:
Use `linter` queue instead of `default` queue for SwiftLint

kean committed:
Update Reader APIs

kean committed:
Add ReaderStream enum

dependabot[bot] committed:
Bump rexml from 3.2.6 to 3.3.9

Bumps [rexml](https://github.com/ruby/rexml) from 3.2.6 to 3.3.9.
- [Release notes](https://github.com/ruby/rexml/releases)
- [Changelog](https://github.com/ruby/rexml/blob/master/NEWS.md)
- [Commits](https://github.com/ruby/rexml/compare/v3.2.6...v3.3.9)

---
updated-dependencies:
- dependency-name: rexml
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

spencertransier committed:
Change swiftlint command

spencertransier committed:
Remove unneeded quotes

hichamboushaba committed:
Add support for sending magic links to inexistent accounts

twstokes committed:
Update changelog.

mokagio committed:
Add PR numbers to public features in changelog

mokagio committed:
Update CocoaPods from 1.15.2 to 1.16.2

This is an attempt to address the 500 error we got from the API when
trying to publish version 17.3.0.

See
https://buildkite.com/automattic/wordpresskit-ios/builds/1908#01934d17-d6b4-43c7-9005-0fcb799f5222

kean committed:
Update Reader APIs (#820)

kean committed:
Add StatsEmailsSummaryData.SortField.postDate

kean committed:
Add StatsEmailsSummaryData.SortField.postDate (#824)

kean committed:
Remove automatically resolved featured images

kean committed:
Fix features images in Reader (#825)

kean committed:
Remove lazy-laod

kean committed:
Task/remove lazy load (#827)

crazytonyli committed:
Add an argument to use ephemeral URLSession to send WP.com API requests

crazytonyli committed:
Update binary target

crazytonyli committed:
Add an argument to use ephemeral URLSession to send WP.com API requests (#828)

kean committed:
Add RemoteComment as a reponse object in CommentServiceRemote

kean committed:
Add autoSuggestedFeaturedImage

kean committed:
Udpate package

kean committed:
Add suitableImageFromPostContent

kean committed:
Update tests

kean committed:
Add auto-suggested featured image back (#831)

crazytonyli committed:
Parse the menu_order property in posts list

crazytonyli committed:
Return visible sites only by default

crazytonyli committed:
Update Package.swift

pmusolino committed:
Add search parameter to `getWPThemesPage` method

pmusolino committed:
update: Package.swift

pmusolino committed:
Revert "update: Package.swift"

This reverts commit a5a49e7bc1b3d15a30ae81254b2ccc504e77e848.

pmusolino committed:
Revert "Add search parameter to `getWPThemesPage` method"

This reverts commit 074fc43d7b404acbe0b481c6d6e3d9e46b162025.

pmusolino committed:
Add search parameter to theme retrieval methods

pmusolino committed:
update: Change WordPressKit binary target URL and checksum

pmusolino committed:
Remove unused filter parameter from ThemeServiceRemote
- Delete `ThemeRequestFilterKey` constant in ThemeServiceRemote.m
- Remove `search` parameter assertion in `getThemesForBlogId` method
- Eliminate filter parameter from `getThemesForBlogId` API call

pmusolino committed:
update `Package.swift`

pmusolino committed:
Change WordPressComRESTAPIVersion from 1_2 to 2_0 in ThemeServiceRemote, `getWPThemesPage` with search method

pmusolino committed:
update package.swift

crazytonyli committed:
Exclude deleted sites from getBlogs (#833)

pmusolino committed:
Merge branch 'wpios-edition' into add-search-param-in-theme-service-remote

pmusolino committed:
update: Package.swift

pmusolino committed:
update: ThemeServiceRemoteTests

pmusolino committed:
update: Package.swift

pmusolino committed:
update: ThemeServiceRemote header file

pmusolino committed:
update: package.swift

pmusolino committed:
fix: endpoint version in method `testThatGetThemesWorks`

pmusolino committed:
update: Package.swift

pmusolino committed:
fix: handle optional search parameter in getWPThemesPage method

pmusolino committed:
update: Package.swift

pmusolino committed:
Add search param in `getWPThemesPage` method and migrate endpoint to v2.0 (#834)

crazytonyli committed:
Exclude unactive Jetpack sites

crazytonyli committed:
Update xcframework url

crazytonyli committed:
Update docs to be consistent with the HTTP API doc

crazytonyli committed:
Update xcframework url

crazytonyli committed:
Do not show disconnected Jetpack sites (#835)

crazytonyli committed:
Update xcframework

crazytonyli committed:
Parse the menu_order property in posts list (#832)

kean committed:
Add /wpcom/v2/sites/:siteid/subscribers

kean committed:
Fix previous Xcode version compatibility

kean committed:
Update tests

kean committed:
Add /wpcom/v2/sites/:siteid/subscribers (#836)

kean committed:
Add search to subscribers

kean committed:
Add search to initializer

kean committed:
Fix missing query

kean committed:
Simplify how filters are passed

kean committed:
Add CaseIterable

kean committed:
Add @frozen

kean committed:
Extract SubscribersServiceRemote

kean committed:
Add getSubsciberStats

kean committed:
Add getSubsciberDetails

kean committed:
Add support for plans

kean committed:
Add SubsciberBasicInfoResponse

kean committed:
Add type parameter

kean committed:
Safer decoding

kean committed:
Cleanup the test data

kean committed:
Make filter public

kean committed:
Add importSubscribers

kean committed:
Fix download link

kean committed:
Extend SubscriberServiceRemote (#837)

mokagio committed:
DRY `env` definition in CI

mokagio committed:
Use latest CI toolkit version

mokagio committed:
AINFRA-283 - Set `queue: mac` explicitly in pipeline

This will allow adopting an improved default pipeline upload step on the
CI infrastructure side of the setup. It also makes it clear what queue
is in use just by looking at the pipeline file.

Notice it does not introduce the standard `shared-pipeline-vars` file
because to use that we first need the default pipeline upload step in
the IaC configuration.

kean committed:
Add search_text parameter to Activity Logs

kean committed:
Update Package.swift

mokagio committed:
AINFRA-283 - Set `queue: mac` explicitly in pipeline (#838)

mokagio committed:
Update README with XCFramework deployment instructions

mokagio committed:
Remove CI steps for CocoaPods – We no longer publish to CocoaPods

kean committed:
Add /rewindable

mokagio committed:
Merge `trunk` + update README in wpios-edition (#840)

mokagio committed:
Add search_text parameter to Activity Logs (#839)

mokagio commented:
In the context of [linear.app/a8c/issue/AINFRA-283](https://linear.app/a8c/issue/AINFRA-283) and as discussed in pbArwn-7sp-p2, I'm going to merge this into https://github.com/wordpress-mobile/WordPressKit-iOS/pull/816 and then https://github.com/wordpress-mobile/WordPressKit-iOS/pull/816 into `trunk`.

In short: WordPress-Jetpack-Reader iOS is now the only client of WordPressKit, so it's okay to have all this on `trunk`. 

We can let the fact the tests don't pass slide for the moment. 

Thanks!

mokagio committed:
Merge remote-tracking branch 'origin/trunk' into wpios-edition
